src/sentry/api/endpoints/organization_events.py

```diff
@@ -88,9 +88,25 @@ def get_v2(self, request, organization):
 
             return self.paginate(
                 request=request,
-                paginator=GenericOffsetPaginator(data_fn=data_fn)
+                paginator=GenericOffsetPaginator(data_fn=data_fn),
+                on_results=lambda results: self.handle_results(request, organization, results),
             )
 
+    def handle_results(self, request, organization, results):
+        projects = {}
+        for project in self.get_projects(request, organization):
```

Comment:
Do we end up fetching the projects twice this way? For larger accounts this could get expensive. Fetching the entire record when we only need id, slug seems wasteful as well.

Reply:
Oof, we weren't.... but you just reminded me we need to do a projects access check here as well!

Reply:
@markstory I updated this to only select the 2 required fields. We do fetch twice, however the first time we fetch the projects is within the `get_filter_params` function which is used in a bunch of places and I'm hesitant to change the signature of that function just for this

Reply:
I agree that changing `get_filter_params` increases risk quite a bit.